packages/ui-causal-graph-editor/src/shared/editor-overlay/legend/legend-data.tsx

```diff
@@ -14,58 +14,98 @@
+}
+
+/**  Union type for the 'type' property */
+type GraphSymbolType = 'none' | 'edge' | 'node';
+
+interface BaseGraphLegendDefinition {
     /** legend label */
     label?: string;
-    /** Show empty spot instead of a legend line */
-    spacer?: boolean;
+    /** legend symbol type */
+    type: GraphSymbolType;
 }
 
-const RESOLVER_LEGEND: LegendLineDefinition[] = [
+// Extend the base interface for each specific 'type' value
+interface NoneGraphLegendDefinition extends BaseGraphLegendDefinition {
+    symbol?: never;
+    type: 'none'; // No symbol for 'none' type
+}
+
+export interface EdgeGraphLegendDefinition extends BaseGraphLegendDefinition {
+    symbol?: LegendArrowDefinition;
+    type: 'edge';
+}
+
+export interface NodeGraphLegendDefinition extends BaseGraphLegendDefinition {
+    symbol?: LegendNodeDefinition;
+    type: 'node';
+}
+
+export type GraphLegendDefinition = NoneGraphLegendDefinition | EdgeGraphLegendDefinition | NodeGraphLegendDefinition;
+
+const RESOLVER_LEGEND: GraphLegendDefinition[] = [
     {
-        centerSymbol: 'question',
-        dashArray: '10 6',
         label: 'Unresolved',
+        symbol: {
+            centerSymbol: 'question',
+            dashArray: '10 6',
+        },
+        type: 'edge',
     },
     {
-        dashArray: '10 6',
         label: 'Not accepted',
+        symbol: {
+            dashArray: '10 6',
+        },
+        type: 'edge',
     },
     {
-        dashArray: '6 4',
         label: 'Accepted',
+        symbol: {
+            dashArray: '6 4',
+        },
+        type: 'edge',
     },
     {
         label: 'Domain knowledge',
+        symbol: {},
+        type: 'edge',
     },
 ];
 
-const PAG_LEGEND: LegendLineDefinition[] = [
-    { arrowType: 'filled', label: 'Directed' },
-    { arrowType: 'empty', label: 'Wildcard' },
-    { arrowType: 'none', label: 'Undirected' },
+const PAG_LEGEND: GraphLegendDefinition[] = [
+    { label: 'Directed', symbol: { arrowType: 'filled' }, type: 'edge' },
+    { label: 'Wildcard', symbol: { arrowType: 'empty' }, type: 'edge' },
+    { label: 'Undirected', symbol: { arrowType: 'none' }, type: 'edge' },
 ];
 
-const ENCODER_LEGEND: LegendLineDefinition[] = [
-    { label: 'Hard Directed' },
-    { arrowType: 'soft', label: 'Soft Directed' },
-    { arrowType: 'none', centerSymbol: 'bidirected', label: 'Undirected' },
-    { arrowType: 'none', centerSymbol: 'cross', label: 'Prohibited' },
+const ENCODER_LEGEND: GraphLegendDefinition[] = [
+    { label: 'Hard Directed', type: 'edge' },
+    { label: 'Soft Directed', symbol: { arrowType: 'soft' }, type: 'edge' },
+    { label: 'Undirected', symbol: { arrowType: 'none', centerSymbol: 'bidirected' }, type: 'edge' },
+    { label: 'Prohibited', symbol: { arrowType: 'none', centerSymbol: 'cross' }, type: 'edge' },
 ];
 
-const DEFAULT_LEGENDS: Map<EditorMode, LegendLineDefinition[]> = new Map([
+const DEFAULT_LEGENDS: Map<EditorMode, GraphLegendDefinition[]> = new Map([
```

Comment:
Should our default legends include the default node color scheme? I believe that was the intention originally but we never got around to doing it

Reply:
there should probably be the default there if the user did not pass anything